lib/medic/store.rb

```diff
@@ -113,5 +115,32 @@ def disable_all_background_delivery(block=Proc.new)
       })
     end
 
+  private
+
+    def prepare_for_save(sample)
+      return sample if sample.kind_of? HKSample
+
+      date = sample[:date] || NSDate.date
+      start_date = sample[:start] || sample[:start_date] || date
+      end_date = sample[:end] || sample[:end_date] || date
+      metadata = sample[:metadata] || {}
```

Comment:
You could continue to use the `sample` hash here rather than local vars, and provide a setup method for it.

```ruby
sample = setup_sample(sample)

# ...

def setup_sample(sample)
  sample = sample.dup
  sample[:date] ||= NSDate.date
  # ...
  sample
end
```